samples/boards/esp32/wifi_apsta_mode/src/main.c

```diff
@@ -0,0 +1,190 @@
+{
+	static struct in_addr addr;
+	static struct in_addr netmaskAddr;
+
+	if (net_addr_pton(AF_INET, CONFIG_WIFI_AP_IP_ADDRESS, &addr)) {
+		LOG_ERR("Invalid address: %s", CONFIG_WIFI_AP_IP_ADDRESS);
+		return;
+	}
+
+	if (net_addr_pton(AF_INET, CONFIG_WIFI_AP_NETMASK, &netmaskAddr)) {
+		LOG_ERR("Invalid netmask: %s", CONFIG_WIFI_AP_NETMASK);
+		return;
+	}
+
+	net_if_ipv4_set_gw(ap_iface, &addr);
+
+	if (net_if_ipv4_addr_add(ap_iface, &addr, NET_ADDR_MANUAL, 0) == NULL) {
+		LOG_ERR("unable to set IP address for AP interface");
+	}
+
+	if (!net_if_ipv4_set_netmask_by_addr(ap_iface, &addr, &netmaskAddr)) {
+		LOG_ERR("Unable to set netmask for AP interface: %s", CONFIG_WIFI_AP_NETMASK);
+	}
+
+	addr.s4_addr[3] += 10; /* Starting IPv4 address for DHCPv4 address pool. */
+
+	if (net_dhcpv4_server_start(ap_iface, &addr) != 0) {
+		LOG_ERR("DHCP server is not started for desired IP");
+		return;
+	}
+
+	LOG_INF("DHCPv4 server started...\n");
+}
+
+static int enable_ap_mode()
+{
+	if (!ap_iface) {
+		LOG_INF("AP: is not initialized");
+		return -EIO;
+	}
+
+	LOG_INF("Turning on AP Mode");
+	ap_config.ssid = (const uint8_t *)CONFIG_WIFI_AP_SSID;
+	ap_config.ssid_length = strlen(CONFIG_WIFI_AP_SSID);
+	ap_config.psk = (const uint8_t *)CONFIG_WIFI_AP_PSK;
+	ap_config.psk_length = strlen(CONFIG_WIFI_AP_PSK);
+	ap_config.channel = WIFI_CHANNEL_ANY;
+	ap_config.band = WIFI_FREQ_BAND_2_4_GHZ;
+
+	if (strlen(CONFIG_WIFI_AP_PSK) == 0) {
+		ap_config.security = WIFI_SECURITY_TYPE_NONE;
+	} else {
+
+		ap_config.security = WIFI_SECURITY_TYPE_PSK;
+	}
+
+	enable_dhcpv4_server();
+
+	int ret = net_mgmt(NET_REQUEST_WIFI_AP_ENABLE, ap_iface, &ap_config,
+			   sizeof(struct wifi_connect_req_params));
+	if (ret) {
+		LOG_ERR("NET_REQUEST_WIFI_AP_ENABLE failed, err: %d", ret);
+	}
+
+	return ret;
+}
+
+static int connect_to_wifi()
+{
+	if (!sta_iface) {
+		LOG_INF("STA: interface no initialized");
+		return -EIO;
+	}
+
+	sta_config.ssid = (const uint8_t *)CONFIG_WIFI_SSID;
+	sta_config.ssid_length = strlen(CONFIG_WIFI_SSID);
+	sta_config.psk = (const uint8_t *)CONFIG_WIFI_PSK;
+	sta_config.psk_length = strlen(CONFIG_WIFI_PSK);
+	sta_config.security = WIFI_SECURITY_TYPE_PSK;
+	sta_config.channel = WIFI_CHANNEL_ANY;
+	sta_config.band = WIFI_FREQ_BAND_2_4_GHZ;
+
+	LOG_INF("Connecting to SSID: %s\n", sta_config.ssid);
+
+	int ret = net_mgmt(NET_REQUEST_WIFI_CONNECT, sta_iface, &sta_config,
+			   sizeof(struct wifi_connect_req_params));
+	if (ret) {
+		LOG_ERR("Unable to Connect to (%s)", CONFIG_WIFI_SSID);
+	}
+
+	return ret;
+}
+
+int main(void)
+{
+	k_sleep(K_SECONDS(5));
+
+	net_mgmt_init_event_callback(&cb, wifi_event_handler, NET_EVENT_WIFI_MASK);
+	net_mgmt_add_event_callback(&cb);
+
+	/* net_if at index 1 is always AP interface in AP-STA mode. */
+	ap_iface = net_if_get_by_index(1);
+
+	/* net_if at index 2 is always STA interface in AP-STA mode. */
+	sta_iface = net_if_get_by_index(2);
+
+	enable_ap_mode();
+	connect_to_wifi();
+
+	while (1) {
```

Comment:
Is this while loop and sleep really needed

Reply:
As variables are defined as global static. It wasn't necessary. We will remove it.